Std/Data/Array/Lemmas.lean

```diff
@@ -721,6 +721,87 @@ theorem all_def {p : α → Bool} (as : Array α) : as.all p = as.data.all p :=
 theorem all_eq_true_iff_forall_mem {l : Array α} : l.all p ↔ ∀ x, x ∈ l → p x := by
   simp only [all_def, List.all_eq_true, mem_def]
 
+/-! ### indexOf? -/
+
+@[simp] theorem indexOf?_data [BEq α] {a : α} {l : Array α} :
+    (l.indexOf? a).map Fin.val = l.data.indexOf? a := by
```

Comment:
Let's not have this be `@[simp]`.

The name is fine, but only if you switch LHS and RHS in the statement. Considering how you use it below, I think that ordering would make more sense.